unittests/test_filters.py

```diff
@@ -140,3 +141,30 @@ def test_invalid_regex(sample_cases):
 
     with pytest.raises(errors.ReframeError):
         count_checks(filters.have_tag('*foo'), sample_cases).evaluate()
+
+
+def test_validates_expr(sample_cases, sample_param_cases):
```

Comment:
Does it make sense to add a `not` test case too? What do you think?

Reply:
Not necessarily as we can't check all valid Python expressions and this is essentially indirectly tested through the `have_cpu` filter.